library/lua/gui/widgets.lua

```diff
@@ -235,9 +235,13 @@ end
 function render_text(obj,dc,x0,y0,pen,dpen,disabled)
     local width = 0
     for iline,line in ipairs(obj.text_lines) do
+        if dc and obj.start_line_num and iline < obj.start_line_num then
```

Comment:
Also, do you need a check for an "end line"?

Reply:
overflow text just isn't displayed. I could have stopped processing it, but I didn't want to complicate the code if I didn't have to. I don't think there is a detectable performance difference. I can add a comment to make this decision explicit, though.

Reply:
Several Painter drawing calls check `isValidPos()`, so they should be at least somewhat faster when drawing text outside the visible region.

Reply:
fair point. done.

Reply:
Oh, I was agreeing with you, not suggesting a change - should have made that clearer. But thanks for the update!